weave/trace/vals.py

```diff
@@ -636,15 +637,21 @@ def make_trace_obj(
     if isinstance(val, ObjectRef):
         new_ref = val
         extra = val.extra
-        read_res = server.obj_read(
-            ObjReadReq(
-                project_id=f"{val.entity}/{val.project}",
-                object_id=val.name,
-                digest=val.digest,
+        try:
+            read_res = server.obj_read(
+                ObjReadReq(
+                    project_id=f"{val.entity}/{val.project}",
+                    object_id=val.name,
+                    digest=val.digest,
+                )
             )
-        )
-        val = from_json(read_res.obj.val, val.entity + "/" + val.project, server)
-        prepare_obj(val)
+            val = from_json(read_res.obj.val, val.entity + "/" + val.project, server)
+            prepare_obj(val)
+        except ObjectDeletedError:
+            # Catch error case where an object has been deleted. Val here is likely
+            # part of a nested object, return None to indicate a deleted object.
```

Comment:
I think it might be safer to have a `DELETED_OBJECT` sentinel.  That makes it very clear to the user that an object used to exist here, but now it doesn't.

The case where `None` is bad is where you have a signature like:
```
class Thing:
    maybe_object: Object | None
```

If you then delete maybe_object, it's ambiguous if `None` means "unset" or "deleted"